compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -59,6 +59,7 @@ fn equate_intrinsic_type<'tcx>(
         require_same_types(
             tcx,
             &cause,
+            ty::ParamEnv::empty(), // FIXME: do all intrinsics have an empty param env?
```

Comment:
Intrinsics can have where clauses (`const_eval_select` is the only one at first glance?)